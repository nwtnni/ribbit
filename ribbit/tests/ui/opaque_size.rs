use core::num::NonZeroU16;

#[derive(Copy, Clone)]
#[ribbit::pack(size = 16, nonzero)]
struct Inner(NonZeroU16);

#[ribbit::pack(size = 21)]
struct Outer {
    pad: u5,
    inner: Inner,
}

fn main() {}
