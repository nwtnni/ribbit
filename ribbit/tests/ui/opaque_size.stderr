error: Opaque type requires size attribute #[ribbit(size = ...)]
  --> tests/ui/opaque_size.rs:10:12
   |
10 |     inner: Inner,
   |            ^^^^^
